internal/db/db_blueprints.go

```diff
@@ -18,6 +19,23 @@ const (
 		INSERT INTO blueprint_versions(id, blueprint_id, version, body)
 		VALUES($1, $2, $3, $4)`
 
+	sqlGetBlueprintComposes = `
+		SELECT composes.job_id, blueprint_versions.version, composes.request, composes.created_at, composes.image_name, composes.client_id
+		FROM composes INNER JOIN blueprint_versions ON composes.blueprint_version_id = blueprint_versions.id
+		WHERE composes.org_id = $1
+		  	AND blueprint_versions.blueprint_id = $2
+			AND CURRENT_TIMESTAMP - composes.created_at <= $3
+			AND ($4::text[] is NULL OR composes.request->'image_requests'->0->>'image_type' <> ALL($4))
+			AND composes.deleted = FALSE
+		ORDER BY composes.created_at DESC
+		LIMIT $5 OFFSET $6`
```

Comment:
Not a huge fan of digging JSON via SQL but you know what, let’s give it a go and this can be always rewritten to client-side filtering OR a new field can be introduced if performance issues arises. Might not be a big deal, SQL optimizer will very likely schedule the scan as the last step.

Reply:
It's copy pasted from fetching composes, might be cood to use some common constant maybe, so once someone gets to refactor it, they'd refactor both :thinking: